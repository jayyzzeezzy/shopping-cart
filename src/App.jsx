import { useState } from "react"
import { Outlet } from "react-router-dom"
import Navbar from "./components/Navbar"

function App() {
  const [cartList, setCartList] = useState([]);

  return (
    <>
      <Navbar />
      
      <Outlet context={[cartList, setCartList]}/>
    </>
  )
}

export default App
