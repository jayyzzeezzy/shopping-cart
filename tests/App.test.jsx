import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import App from "../src/App";

describe("App component", () => {
  it("renders snapshot", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  })

  it("renders correct heading", () => {
    render(<App />);
    expect(screen.getByRole("heading", { name: "THE ONLINE SHOP" }));
  });
});