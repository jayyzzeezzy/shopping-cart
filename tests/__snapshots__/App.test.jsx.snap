// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders snapshot 1`] = `
<div>
  <h1>
    THE ONLINE SHOP
  </h1>
</div>
`;
